docs/api/effector/createEffect.md

```diff
@@ -4,55 +4,116 @@ title: createEffect
 
 #### Returns
 
 ([_`Effect`_](Effect.md)): A container for async function.
 
-#### Example
+#### Examples
+
+Create unnamed effect
 
 ```js
-const getUser = createEffect('get user', {
-  handler: params => fetch(`https://example.com/get-user/${params.id}`)
-    .then(res => res.json())
+import {createEffect} from 'effector'
+
+const fetchUserRepos = createEffect({
+  handler: ({name}) => {
+    return fetch(`https://api.github.com/users/${user}/repos`)
+      .then(res => res.json())
+  }
+})
+```
+
+Create named effect
+
+```js
+import {createEffect} from 'effector'
+
+const fetchUserRepos = createEffect('fetch user repositories', {
+  handler: ({name}) => {
+    return fetch(`https://api.github.com/users/${user}/repos`)
+      .then(res => res.json())
+  }
 })
+```
+
+Set handler to effect after creating
+
+```js
+import {createEffect} from 'effector'
+
+const fetchUserRepos = createEffect()
 
-// OR
-getUser.use(params => {
-  return fetch(`https://example.com/get-user/${params.id}`)
+fetchUserRepos.use(({name}) => {
+  return fetch(`https://api.github.com/users/${user}/repos`)
     .then(res => res.json())
 })
+```
 
-const users = createStore([]) // <-- Default state
-  // add reducer for getUser.done event (fires when promise resolved)
-  .on(getUser.done, (state, {result: user, params}) => [...state, user])
+Watch effect status
 
-// subscribe to promise resolve
-getUser.done.watch(({result, params}) => {
-  console.log(params) // {id: 1}
+```js
+import {createEffect} from 'effector'
+
+const fetchUserRepos = createEffect({
+  handler: ({name}) => {
+    return fetch(`https://api.github.com/users/${name}/repos`)
+      .then(res => res.json())
+  }
+})
+
+fetchUserRepos.pending.watch(pending => {
+  return console.log(`effect is pending?: ${pending ? 'yes' : 'no'}`)
+})
+
+fetchUserRepos.done.watch(({params, result}) => {
+  console.log(params) // {name: 'zerobias'}
   console.log(result) // resolved value
 })
 
-// subscribe to promise reject (or throw)
-getUser.fail.watch(({error, params}) => {
-  console.error(params) // {id: 1}
+fetchUserRepos.fail.watch(({params, error}) => {
+  console.error(params) // {name: 'zerobias'}
   console.error(error) // rejected value
 })
 
-// you can replace function anytime
-getUser.use(() => promiseMock)
+fetchUserRepos.finally.watch(({ params, status, result, error }) => {
+  console.log(params) // {name: 'zerobias'}
+  console.log(`handler status: ${status}`)
+  
+  if (error) {
+    console.log('handler rejected', error)
+  } else {
+    console.log('handler resolved', result)
+  }
+})
+
+fetchUserRepos({name: 'zerobias'})
+```
 
-// call effect with your params
-getUser({id: 1})
+Change state
+
+```js
+import {createStore, createEffect} from 'effector'
+
+const fetchUserRepos = createEffect({
+  handler: ({name}) => {
+    return fetch(`https://api.github.com/users/${name}/repos`)
+      .then(res => res.json())
```

Comment:
Maybe it is better to use async/await here? It could be more readable

Reply:
> Maybe it is better to use async/await here? It could be more readable

like this?

```
const fetchUserRepos = createEffect({
  handler: async ({name}) => {
  	try {
  		const response = await fetch(`https://api.github.com/users/${name}/repos`)
  		const json = await response.json()
    	return json
  	} catch(error) {
    	return error
  	}
	}
})
```

Reply:
We don't need no try catch inside handler, that's the point of effects: we always can handle errors by `fetchUserRepos.fail`

```javascript
const fetchUserRepos = createEffect({
  async handler({name}) {
    const url = `https://api.github.com/users/${name}/repos`
    const req = await fetch(url)
    return req.json()
  }
})
```

Reply:
> We don't need no try catch inside handler, that's the point of effects: we always can handle errors by `fetchUserRepos.fail`
> 
> ```js
> const fetchUserRepos = createEffect({
>   async handler({name}) {
>     const url = `https://api.github.com/users/${name}/repos`
>     const req = await fetch(url)
>     return req.json()
>   }
> })
> ```

Great!